src/components/Notifications/ToastNotification/ToastNotificationProvider.tsx

```diff
@@ -25,6 +25,7 @@ export type ToastNotificationType = NotificationType & {
 
 interface Props {
   onDismiss?: (notifications?: ToastNotificationType[]) => void;
+  hideDelay?: number | false;
```

Comment:
Solely my opinion: I am not a fan of those multi type props. How about using a special value to deactivate it, like setting it to 0?